libsql/internal/http/hranaV2/hranaV2.go

```diff
@@ -286,10 +287,120 @@ func sendPipelineRequest(ctx context.Context, msg *hrana.PipelineRequest, url st
 	return result, false, nil
 }
 
+func (h *hranaV2Conn) executeMsg(ctx context.Context, msg *hrana.PipelineRequest) (*hrana.PipelineResponse, error) {
+	result, err := h.sendPipelineRequest(ctx, msg, false)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range result.Results {
+		if r.Error != nil {
+			return nil, errors.New(r.Error.Message)
+		}
+		if r.Response == nil {
+			return nil, errors.New("no response received")
+		}
+	}
+	return result, nil
+}
+
+type chunker struct {
+	chunk    []string
+	iterator *sqliteparserutils.StatementIterator
+	limit    int
+}
+
+func newChunker(iterator *sqliteparserutils.StatementIterator, limit int) *chunker {
+	return &chunker{iterator: iterator, chunk: make([]string, 0, limit), limit: limit}
+}
+
+func (c *chunker) Next() (chunk []string, isEOF bool) {
+	c.chunk = c.chunk[:0]
+	var stmt string
+	for !isEOF && len(c.chunk) < c.limit {
+		stmt, _, isEOF = c.iterator.Next()
+		// We need to skip transaction statements. Chunks run in a transaction by default.
+		if stmt != "" && !strings.HasPrefix(stmt, "BEGIN") && !strings.HasPrefix(stmt, "COMMIT") && !strings.HasPrefix(stmt, "END") && !strings.HasPrefix(stmt, "ROLLBACK") {
```

Comment:
Don't remember exactly how statements splitting works - do we trim statement strings?
Also, should we perform non-case-sensitive comparison here?

Reply:
Also, triggers emitted as one single statement, right? (because transaction inside trigger is fine)

Reply:
This is a good point about case-sensitivity. I will test the whitespaces and triggers as well. Thanks

Reply:
Spaces are trimmed.